common/scala/src/main/scala/whisk/core/entity/WhiskEntity.scala

```diff
@@ -60,11 +60,17 @@ abstract class WhiskEntity protected[entity] (en: EntityName) extends WhiskDocum
   /** The primary key for the entity in the datastore */
   override final def docid = fullyQualifiedName(false).toDocId
 
+  /** Type of entity */
+  def entityType: String
+
   /**
    * Returns a JSON object with the fields specific to this abstract class.
    */
   protected def entityDocumentRecord: JsObject =
-    JsObject("name" -> JsString(name.toString), "updated" -> JsNumber(updated.toEpochMilli()))
+    JsObject(
+      "name" -> JsString(name.toString),
+      "updated" -> JsNumber(updated.toEpochMilli()),
+      "type" -> JsString(entityType))
```

Comment:
I’d use the property kind instead of type, or entityType. The string type could cause problems in the future because it’s a reserved string in scala and would have to be escaped in a case class for example with default serdes.

Reply:
Ack. Used `entityType` as property name as `kind` is already being used for `exec` to avoid confusuion